### PR TITLE
Usage of dynamic dots in `router_ui`

### PR DESCRIPTION
**Pull request by vanhry**
### Changes
- Change dots `...` argument in `router_ui` to allow dynamically pass arguments
- According test was added
- One example of usage was added

### CODE REVIEW

File: R/router.R [hunk 1]
```diff
@@ -203,9 +205,36 @@ create_router_callback <- function(root, routes = NULL) {
 #'     )
 #'   }
 #' }
+#' \dontrun{
+#'   dynamic_routes <- list(
+#'     route("/other2", other_page(id = "other2")),
+#'     route("/other3", other_page(id = "other3"))
```
Comment (jakubnowicki) [inline R/router.R:211]:
```suggestion
#'     route("other2", other_page(id = "other2")),
#'     route("other3", other_page(id = "other3"))
```

File: R/router.R [hunk 2]
```diff
@@ -203,9 +205,36 @@ create_router_callback <- function(root, routes = NULL) {
...
+#'   )
+#'
+#'   ui <- function() {
+#'     router_ui(
+#'       route("/", root_page(id = "root")),
+#'       route("/other", other_page(id = "other")),
```
Comment (jakubnowicki) [inline R/router.R:217]:
```suggestion
#'       route("other", other_page(id = "other")),
```

File: R/router.R [hunk 3]
```diff
@@ -182,7 +182,9 @@ create_router_callback <- function(root, routes = NULL) {
 #' Creates router UI in Shiny applications.
 #'
 #' @param default Main route to which all invalid routes should redirect.
-#' @param ... All other routes defined with shiny.router::route function.
+#' @param ... <[`dynamic-dots`][rlang::dyn-dots]>
+#'   All other routes defined with shiny.router::route function.
+#'   It's possible to pass routes in dynamic way with dynamic dots
```
Comment (jakubnowicki) [inline R/router.R:187]: Please remove the beginning (`<[`dynamic-dots`][rlang::dyn-dots]>`). Instead, after mentioning the possibility to use dynamic dots please:
- provide a link to `rlang` documentation with dynamic dots explanation
- mention the example

File: R/router.R [hunk 4]
```diff
@@ -203,9 +205,36 @@ create_router_callback <- function(root, routes = NULL) {
 #'     )
 #'   }
 #' }
+#' \dontrun{
+#'   dynamic_routes <- list(
```
Comment (jakubnowicki) [inline R/router.R:209]: Please add an example application that uses dynamic dots to `examples` directory.

File: R/router.R [hunk 5]
```diff
@@ -203,9 +205,36 @@ create_router_callback <- function(root, routes = NULL) {
 #'     )
 #'   }
 #' }
+#' \dontrun{
```
Comment (jakubnowicki) [inline R/router.R:208]: Please add a minimal description to this example.

File: inst/WORDLIST [hunk 6]
```diff
@@ -24,3 +24,5 @@ subpages
 ui
 webapp
 www
+rlang
```
Comment (jakubnowicki) [inline inst/WORDLIST:27]: Please use alphabetical order.

File: tests/testthat/test-router.R [hunk 7]
```diff
@@ -43,3 +43,49 @@ test_that("test getting clean url hash", {
 
   expect_equal(shiny::isolate(get_url_hash(session)), "#!/")
 })
+
+
+test_that("test router_ui with dynamic dots", {
+  default <- route("/", shiny::div("default"))
+  first <- route("/first", shiny::div("first"))
+  second <- route("/second", shiny::div("second"))
```
Comment (jakubnowicki) [inline tests/testthat/test-router.R:51]:
```suggestion
  first <- route("first", shiny::div("first"))
  second <- route("second", shiny::div("second"))
```

File: tests/testthat/test-router.R [hunk 8]
```diff
@@ -43,3 +43,49 @@ test_that("test getting clean url hash", {
...
+  router_ui_dynamic <- router_ui(default = default, !!!list(first, second))
+  router_ui_static <- router_ui(default = default, first, second)
+
+  expect_identical(router_ui_dynamic, router_ui_static)
+})
+
+test_that("named args in dynamic dots are ignored", {
+  default <- route("/", shiny::div("default"))
+  first <- route("/first", shiny::div("first"))
+  second <- route("/second", shiny::div("second"))
```
Comment (jakubnowicki) [inline tests/testthat/test-router.R:62]:
```suggestion
  first <- route("first", shiny::div("first"))
  second <- route("second", shiny::div("second"))
```

File: DESCRIPTION [hunk 9]
```diff
@@ -23,7 +23,8 @@ License: MIT + file LICENSE
 Imports:
     htmltools,
     glue,
-    shiny
+    shiny,
+    rlang
```
Comment (jakubnowicki) [inline DESCRIPTION:27]: Please use alphabetical order.

**Review by jakubnowicki — CHANGES_REQUESTED** [review]
Overall, this is a nice improvement to the package. Some changes are required, but rather minor ones. Also, please update the package version in the DESCRIPTION.

File: R/router.R (hunk 3 above)
Comment (vanhry) [reply R/router.R]: Done, should I somehow add the link to the file in `examples/`

File: R/router.R (hunk 4 above)
Comment (vanhry) [reply R/router.R:211]: Added few comments

File: R/router.R (hunk 5 above)
Comment (vanhry) [reply R/router.R:209]: Added few comments

**Review by jakubnowicki — APPROVED** [review]
LGTM

**Pull request merged**